solutions/USACO open18-familytree.mdx

```diff
@@ -0,0 +1,127 @@
+---
+id: usaco-833
+source: USACO Bronze 2018 Open
+title: Family Tree
+author: Danh Ta Chi Thanh
+---
+
+## Solution
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_family_bronze_open18.html)
+
+## Alternative Solution - BFS

```

Comment:
I forgot to mention.. what is runtime?

Reply:
It's O(N^2), but can be faster by using `map` to store the value.

Reply:
Yea, I'd like to see it faster -- it feels odd for a bottleneck to be so easily avoidable...

Reply:
The constraints are low, so I don't know if the constant factors from inserting and finding elements in a `std::map` would be an issue or not.

Reply:
The point is that if we want a overkill solution, why have it only be partly overkill? The fact that it's kind of in the middle makes it un-understandable for beginners (which is intended) but also not as useful to those more familiar w the concept.

Reply:
I think the BFS solution should be just for those who want to use a different approach for this problem.